ui/packages/atlasmap-core/src/services/mapping-expression.service.spec.ts

```diff
@@ -168,4 +174,59 @@ describe('MappingExpressionService', () => {
     const mapping1 = cfg.mappings!.mappings[1];
     expect(service.createMappingExpression(mapping1)).toContain('Concatenate');
   });
+
+  test('test adding a field to an expression makes it into the mapping', () => {
+    TestUtils.createMockMappings(cfg);
+    const mapping = cfg.mappings!.mappings[1];
+
+    const docDef = new DocumentInitializationModel();
+    docDef.type = DocumentType.JSON;
+    docDef.inspectionType = InspectionType.SCHEMA;
+    docDef.name = 'JSONSchemaSource';
+    docDef.isSource = true;
+    docDef.id = 'JSONSchemaSource';
+    docDef.description = 'random desc';
+    const jsonSchemaSource = cfg.addDocument(docDef);
+
+    const source = new Field();
+    source.isPrimitive = true;
+    source.path = '/addressList<>/city';
+    source.docDef = jsonSchemaSource;
+    source.parentField = new Field();
+    source.parentField.isCollection = true;
+    source.parentField.parentField = new Field();
+    source.parentField.parentField.isCollection = true;
+    jsonSchemaSource.addField(source);
+
+    expect(mapping).toBeDefined();
+    cfg.mappingService.selectMapping(mapping);
+
+    const textNode: ExpressionNode = {
```

Comment:
This could be
```javascript
const textNode: ExpressionNode = {
      uuid: '0',
      str: 'mockstr',
      getUuid: () => '',
      toText: () => '',
      toHTML: () => '',
    };
```

Reply:
sure - np